backend/geohashing.js

```diff
@@ -0,0 +1,115 @@
+    const latitude = Math.random() * 180 - 90;
+    const longitude = Math.random() * 360 - 180;
+    return { latitude, longitude };
+}
+
+function generateRandomCoordinates(coordinates_num) {
+    const coordinatesArray = [];
+    for (let i = 0; i < coordinates_num; i++) {
+        const coordinates = generateRandomElement();
+        coordinatesArray.push(coordinates);
+    }
+    return coordinatesArray;
+}
+
+function encodeGeohash(lat, long, precision = 12) {
+    const latInterval = [-90, 90];
+    const longInterval = [-180, 180];
+    let geohash = '';
+    const base32Chars = '0123456789bcdefghjkmnpqrstuvwxyz';
+    let isEvenBit = true;
+
+    while (geohash.length < precision) {
+        let mid;
+        let charBits = 0;
+        for (let bit = 4; bit >= 0; --bit) {
+            if (isEvenBit) {
+                mid = (longInterval[0] + longInterval[1]) / 2;
+                if (long > mid) {
+                    charBits |= 1 << bit;
+                    longInterval[0] = mid;
+                } else {
+                    longInterval[1] = mid;
+                }
+            } else {
+                mid = (latInterval[0] + latInterval[1]) / 2;
+                if (lat > mid) {
+                    charBits |= 1 << bit;
+                    latInterval[0] = mid;
+                } else {
+                    latInterval[1] = mid;
+                }
+            }
+            isEvenBit = !isEvenBit;
+        }
+        geohash += base32Chars[charBits];
+    }
+    return geohash;
+}
+
+function geohashes(coordinates, min_long, max_long, min_lat, max_lat, geohash) {
+    if (coordinates.length < 100) {
+        return coordinates.map(coord => {
+            const lat = coord.latitude;
+            const long = coord.longitude;
+            const geoHash = encodeGeohash(lat, long);
+            return { ...coord, geohash: geoHash };
+        });
+    } else {
+        return subdivide(coordinates, min_long, max_long, min_lat, max_lat, geohash);
+    }
+}
+
+function subdivide(coordinates, min_long, max_long, min_lat, max_lat, geohash) {
+    const midLat = (min_lat + max_lat) / 2;
+    const midLong = (min_long + max_long) / 2;
+
+    const topLeft = coordinates.filter(c => c.latitude >= midLat && c.longitude < midLong);
+    const topLeftResult = geohashes(topLeft, min_long, midLong, midLat, max_lat, geohash + '0');
+
+    const topRight = coordinates.filter(c => c.latitude >= midLat && c.longitude >= midLong);
+    const topRightResult = geohashes(topRight, midLong, max_long, midLat, max_lat, geohash + '1');
+
+    const bottomLeft = coordinates.filter(c => c.latitude < midLat && c.longitude < midLong);
+    const bottomLeftResult = geohashes(bottomLeft, min_long, midLong, min_lat, midLat, geohash + '2');
+
+    const bottomRight = coordinates.filter(c => c.latitude < midLat && c.longitude >= midLong);
+    const bottomRightResult = geohashes(bottomRight, midLong, max_long, min_lat, midLat, geohash + '3');
```

Comment:
Order of arguments is off here too: can you take another look and amend a new change?